usr/local/share/bastille/etcupdate.sh

```diff
@@ -0,0 +1,254 @@
+
+EOF
+        exit 1
+}
+
+
+executeconditional() {
+    if [ $DRY_RUN -eq "0" ]; then
+        eval "$@"
+    fi
+}
+
+C1_C6_conditions() {
+    filelistjail=$(find "${jail_etc}" -mindepth 1 -type f)
+    for jailfile in ${filelistjail}
+    do
+        filepart=$(echo "${jailfile}" | awk -F 'etc/' '{print $NF}')
+        newbasefile="${new_basedir}/etc/${filepart}"
+        currentbasefile="${current_basedir}/etc/${filepart}"
+
+        if [ ! -f "${currentbasefile}" ]; then
+            if [ ! -f "${newbasefile}" ]; then
+                C1=$((C1+1))
+                C1ct="$C1ct${jailfile}\n"
+            else
+                C2=$((C2+1))
+                C2ct="$C2ct${jailfile}\n"
+            fi
+        fi
+
+        if [ -f "${currentbasefile}" ]; then
+            diffr=$(diff -u "${jailfile}" "${currentbasefile}")
+            if [ -z "${diffr}" ]; then
+                if [ ! -f "${newbasefile}" ]; then
+                    C3=$((C3+1))
+                    C3ct="$C3ct${jailfile}\n"
+                    cmd="rm -rf ${jailfile}"
+                    executeconditional "$cmd"
+                else
+                    C4=$((C4+1))
+                    C4ct="$C4ct${jailfile}\n"
+                    cmd="cp -p ${newbasefile} ${jailfile}"
+                    executeconditional "$cmd"
+                    # Copy keeping permissions
+                fi
+            else
+                diffs="${diffs}${diffr}"
+                diffs="${diffs}\n==========================================================================================================\n\n"
+                if [ -f "${newbasefile}" ]; then
+                    C5=$((C5+1))
+                    C5ct="$C5ct${jailfile}\n"
+                else
+                    C6=$((C6+1))
+                    C6ct="$C6ct${jailfile}\n"
+                fi
+            fi
+        fi
+    done
+}
+
+# Creates missing directories from UPGRADEVERSION in the jail preserving original permissions
+C7_conditions() {
+dirlistrelease=$(find "${new_basedir}/etc" -mindepth 1 -type d)
+    for dirpath in ${dirlistrelease}
+    do
+        dirpathnf=$(echo "${dirpath}" | awk -F '/etc' '{print $NF}')
+        jailpath="${bastille_jail_base}/root/etc${dirpathnf}"
+        if [ ! -d "${jailpath}" ]; then
+            C7=$((C7+1))
+            cmd="mkdir ${jailpath}"
+            executeconditional "$cmd"
+            dirperm=$(stat -f "%Mp%Lp" "${dirpath}")
+            cmd="chmod ${dirperm} ${jailpath}"
+            executeconditional "$cmd"
+            C7ct="$C7ct${jailpath}\n"
+        fi
+    done
+}
+
+# Copy missing files from UPGRADEVERSION to the jail preserving original permissions
+C8_conditions() {
+filelistrelease=$(find "${new_basedir}/etc" -mindepth 1 -type f)
+    for sourcefile in ${filelistrelease}
+    do
+        dirpathnf=$(echo "${sourcefile}" | awk -F '/etc' '{print $NF}')
+        jailfile="${bastille_jail_base}/root/etc${dirpathnf}"
```

Comment:
Suggest changing `bastille_jail_base` to something like `bastille_thin_jail` because the former makes it sound like a "base" jail, which is what the release jails are in this context.  The thin jail is the jail to be operated on.